ruby_programming/files_and_serialization/lesson_serialization.md

```diff
@@ -36,3 +36,11 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Zetcode's section on Input/Output in Ruby](http://zetcode.com/lang/rubytutorial/io/) should be another useful perspective on the material.
 * [Ruby Monk's section on Serializing](https://web.archive.org/web/20160505174806/http://rubymonk.com/learning/books/4-ruby-primer-ascent/chapters/45-more-classes/lessons/104-serializing)
 * [Short Example of Serialization](http://rubylearning.com/satishtalim/object_serialization.html) from Ruby Learning
+
+### Knowledge Check
+
+  * <a class='knowledge-check-link' href='https://thoughtbot.com/blog/io-in-ruby'>What does the IO object class do?</a>
+  * <a class='knowledge-check-link' href='https://www.youtube.com/watch?v=uS37TujnLRw'>How are data structures related to serialization?</a>
+  * <a class='knowledge-check-link' href='https://www.sitepoint.com/choosing-right-serialization-format/'>What are three different serialization formats?</a>
+  * <a class='knowledge-check-link' href='https://www.sitepoint.com/choosing-right-serialization-format/'>What are the benefits of each type of serialization mentioned?</a>
```

Comment:
I would like to see this question rewritten to not depend don't the context of the previous question. I.e. something along the lines of "benefits of YAML vs JSON vs MessagePack"

Reply:
Rewrote it to make it not reliant on the previous question. I thought it would be better to include serializations format at the end to make it clear what it's referring to without any context from the previous question.

Reply:
In some cases it can be more about the implementation of a concept rather than a concept itself, but because that isn't always the case that phrasing wouldn't fit in something general purpose like the wiki.